src/connector/src/sink/kafka.rs

```diff
@@ -439,15 +452,19 @@ impl KafkaPayloadWriter {
                 Err((e, rec)) => {
                     record = rec;
                     match e {
-                        err @ KafkaError::MessageProduction(RDKafkaErrorCode::QueueFull)
-                        | err @ KafkaError::MessageProduction(RDKafkaErrorCode::MessageTimedOut) => {
```

Comment:
We read the librdkafka code and realize that MessageTimeOut won't happen in `send_result`. It can happen when awaiting the delivery future and when we see that error in delivery future, there is nothing much we can do other than throwing out the error and triggering recovery.